projects/igniteui-angular/src/lib/badge/badge.component.ts

```diff
@@ -169,7 +179,8 @@ export class IgxBadgeComponent {
      * @hidden
      * @internal
      */
-    public setClasses() {
+    @HostBinding('class')
```

Comment:
We should rather bind each class separately. This will require to split this function into several smaller getters that assert the type and bind only the specific class. This way any custom classes added to the host element by the end-user will not be overwritten. Another solution would be to follow the pattern described in [this issue](https://github.com/angular/angular/issues/7289#issuecomment-390415535).